### PR TITLE
Fix a reported bug in fw_toggle

### PR DESCRIPTION
**Pull request by kelvin-cao**
In fw_toggle(), after the toggle operation, it will get fw info and display. In the case of a failed toggle operation, it is observed that the fw-toggle command still show suceess after the fw info display. 

The reason is that when get fw info, the switchtec_fw_part_info(), which is called when getting the fw info, will clear the errno in some cases when it ends up succeeded. This patch remove the errno clear code to make sure it's untouched on success return of the function.

### CODE REVIEW

**Comment by lsgunth** [comment]
First, can you get in the habit of putting the diagnosis in the commit message? Not in the PR. That's where we need the information.

Second I like this solution. I think it is an issue in the fw-toggle command. Calling a function that causes an error then, doing something else and hopping the error is the same is probably broken. We should fix that.

**Comment by kelvin-cao** [comment]
For the fist, I originally thought the previous commit was a general fix which is not necessarily link to this issue, so I didn't put the message there. For the second, I agree. 

I've updated with two commits. Thanks.

**Comment by kelvin-cao** [comment]
@lsgunth Kindly have a look at the update. Thanks.

File: lib/fw.c [hunk 1]
```diff
@@ -1000,7 +1000,6 @@ static int switchtec_fw_part_info(struct switchtec_dev *dev, int nr_info,
 			return ret;
 
 		if (ret) {
-			errno = 0;
```
[the remark this answers is not in the document]
Comment (kelvin-cao) [reply lib/fw.c]: Below is what I found in the errno man page. I think we can follow this practice. However, it doesn't make any difference with current error handling we have. 

> Valid error numbers are all nonzero; errno is never set to zero by any system call or library function.

File: cli/main.c [hunk 2]
```diff
@@ -1425,11 +1426,16 @@ static int fw_toggle(int argc, char **argv)
 							   cfg.key,
 							   cfg.firmware,
 							   cfg.config);
+		err = errno;
```
[the remark this answers is not in the document]
Comment (kelvin-cao) [reply cli/main.c]: I think showing the partition info even after a failed toggle is helpful to user to understand current situation.

File: cli/main.c [hunk 3]
```diff
@@ -1425,11 +1426,16 @@ static int fw_toggle(int argc, char **argv)
...
 
-	print_fw_part_info(cfg.dev);
+	ret = print_fw_part_info(cfg.dev);
+	if (ret)
+		switchtec_perror("print fw info");
+
 	printf("\n");
 
+	errno = err;
 	switchtec_perror("firmware toggle");
```
[the remark this answers is not in the document]
Comment (kelvin-cao) [reply cli/main.c]: Since I want to show partition info even after a failed toggle, so this might show failure info here.

**Pull request merged**